### ISSUE TITLE
[Protocol] MVP implementation off the on-chain `SubmitProof` message handling

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/poktroll#141]
## Objective

MVP Implementation of `SubmitProof` to enable Shannon TestNet.

## Origin Document

- #44

## Goals

- Enable E2E validation of the entire protocol
- Unblock iterative development of the claim & proof lifecycle
- Unblock iterative development of on-chain tokenomics

## Deliverables

- [ ] A single PR that introduces the business logic for the on-chain `SubmitProof` message (verification, etc...)
- [x] A single PR that updates the CLI to submit SubmitProof messages
- [ ] Introduce make targets to enable development (visibility, triggering, debugging , etc...) of on-chain proofs
- [ ] Add extensive unit tests 

## Non-goals / Non-deliverables

- Account Management (i.e. Burn & Mint) business logic related to the Claim & Proof lifecycle (leave TODOs)
- Governance Params necessary for the Claim & Proof lifecycle (leave TODOs)

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @Olshansk 
**Co-Owners**: @bryanchriswhite

**Comment by bryanchriswhite** [comment]
@Olshansk I'm starting on; got a happy path unit test already.
> A single PR that introduces the business logic for the on-chain SubmitProof message (verification, etc...)

**Comment by Olshansk** [comment]
@bryanchriswhite Have you (planning to, considering, open-to) implemented the equivalent of #148 and #151 for proofs?

I created #243 for the work we discussed today and will drive that myself but wanted to know if the above is on your radar.

**Comment by bryanchriswhite** [comment]
> @bryanchriswhite Have you (planning to, considering, open-to) implemented the equivalent of #148 and #151 for proofs?

Yessir! The starting point I was referring to is analogous to #148 + #236 - testutils (which I will endeavor to isolate as reasonable). I think this is the best place to start as it's the most "outside", allowing me to work outside-in. Parts of the #151 analogue may be driven out along the way but otherwise I think it will be quite natural to lay out more isolated PRs for the various pieces.

> I created #243 for the work we discussed today and will drive that myself but wanted to know if the above is on your radar.

:+1: :pray:
